ui/dashboards/src/components/PanelDrawer/panel-editor-model.ts

```diff
@@ -0,0 +1,149 @@
+// Copyright 2022 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { useMemo } from 'react';
+import { usePlugin } from '@perses-dev/plugin-system';
+import { useImmer } from 'use-immer';
+import { useDashboardApp, usePanels } from '../../context';
+import { removeWhiteSpacesAndSpecialCharacters } from '../../utils/functions';
+
+export interface PanelEditorFormValues {
+  name: string;
+  description: string;
+  group: number;
+  kind: string;
+  spec: unknown;
+}
+
+// Props that vary based on whether the PanelDrawer is in add or edit mode
+interface PanelDrawerModel {
+  drawerTitle: string;
+  submitButtonText: string;
+  initialValues: PanelEditorFormValues;
+  applyChanges: (values: PanelEditorFormValues) => void;
+}
+
+/**
```

Comment:
Nice! I love all the comments here and below, makes it really clear what's happening